doc/topics/the-event-loop-timers-and-nexttick.md

```diff
@@ -0,0 +1,373 @@
+# The Node.js Event Loop, Timers, and `process.nextTick()`
+
+## What is the Event Loop?
+
+The event loop is what allows Node.js to perform non-blocking I/O
+operations — despite the fact that JavaScript is single-threaded — by
+offloading operations to the system kernel whenever possible. 
+
+Since most modern kernels are multi-threaded, they can handle
+multiple operations executing in the background. When one of these
+operations completes, the kernel tells Node.js so that the appropriate callback 
+may added to the `poll` queue to eventually be executed. We'll explain
+this in further detail later in this topic. 
+
+## Event Loop Explained 
+
+When Node.js starts, it initializes the event loop, processes the provided
+input script (or drops into the REPL, which is not covered in this document)
+which may make async API calls, schedule timers, or call `process.nextTick()`, 
+then begins processing the event loop. 
+
+The following diagram shows a simplified overview of the event loop's
+order of operations.
+
+       ┌───────────────────────┐
```

Comment:
The order in which certain types of callbacks are invoked is mostly determined by libuv, see https://github.com/libuv/libuv/blob/b015b4d0ee03b825d61b0b54e36aa405a8e1cf03/src/unix/core.c#L340-L366.

My take would be:

```
1. timers
2. i/o callbacks (pretty much every type of callback except timers, setImmedate, close)
2. idle, prepare (only used internally)
2. poll (= retrieve new i/o events; node will block here when appropriate)
3. check (setImmediate callbacks are invoked here)
4. close callbacks (e.g `socket.on('close', ...)`)
5. timers (again, indeed)
```

process.nextTick and microtasks are processed after every invocation of a javascript function, so they essentially run as a "loop within a loop" with a higher priority.